feat(preview): make UI changes and allow individual dimensions to be editable

gargi-bansal opened the pull request:
### Purpose

User can now edit slider in preview page and click on dimension title to go to fully edit the dimension

[HRT-62](https://softeng761team5.atlassian.net/browse/HRT-62)

### Approach
Opens up a new component that looks identical to Display Cards to allow for full editing

### Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

<!--
If this PR contains a breaking change, describe the impact and migration path for existing applications below.
-->

### Screenshots before and after this change (required for UI changes)

#### Does this PR involve a UI change?

- [x] Yes
- [ ] No

#### Screenshot(s) of UI before this PR
N/A

#### Screenshot(s) of UI after this PR
<img width="1000" alt="Screen Shot 2020-10-01 at 7 38 57 PM" src="https://user-images.githubusercontent.com/42624632/94776208-ce802780-041d-11eb-8374-37b511271a8d.png">
<img width="1000" alt="Screen Shot 2020-10-01 at 7 38 46 PM" src="https://user-images.githubusercontent.com/42624632/94776217-d049eb00-041d-11eb-9381-992578277682.png">

gargi-bansal committed:
feat(front end): made changes to preview page UI to clean up and redirect to Display cards

made changes to preview page UI to clean up and redirect to Display cards

gargi-bansal committed:
Merge branch 'master' into hrt-62-preview-ui-changes

gargi-bansal committed:
feat(front end): preview page opens individual dimension for editing

Preview page opens individual dimension for editing

gargi-bansal committed:
Merge branch 'master' into hrt-62-preview-ui-changes

gargi-bansal committed:
feat(front end): fixed preview styling due to merge changes

fixed preview styling due to merge changes

gargi-bansal committed:
feat(front end): added hover to link on preview

Added hover to link on preview

mitchfaulcon reviewed (commented):
Looks good, but just checking:
 - Should the right footer button still say save when you've clicked back to a dimension from the preview page?
- Should clicking back on the preview page take you all the way back to the start or just back to the last dimension? At the moment you go all the way back to the start and have to click through them all again

gargi-bansal commented:
> Looks good, but just checking:
> 
> * Should the right footer button still say save when you've clicked back to a dimension from the preview page?
> * Should clicking back on the preview page take you all the way back to the start or just back to the last dimension? At the moment you go all the way back to the start and have to click through them all again

I agree with the first one, I couldn't figure out what to name the buttons on the individual preview editing.
I think I have an idea to fix the second bullet point so I'll do it now.

gargi-bansal committed:
feat(preview): preview page takes user back to last dimension on display cards page

preview page takes user back to last dimension on display cards page

gargi-bansal committed:
feat(front end): button width changes for longer button text, need to assigned additional class name

button width changes for longer button text, need to assigned additional class name

gargi-bansal committed:
feat(front end): dimension changes are not saved on cancel

Dimension changes are not saved on cancel

mitchfaulcon commented on src/components/PreviewDimension.tsx:
```diff
@@ -0,0 +1,264 @@
+export enum CardSide {
+  Left,
+  Right,
+}
```
I think this should be imported from DisplayCards instead defined again

gargi-bansal replied:
okay, I'll just refactor out all the common things tomorrow then!

gargi-bansal committed:
refactor(front end): refactored common code between DisplayCards and PreviewDimensions

Refactored common code between DisplayCards and PreviewDimensions

gargi-bansal committed:
feat(front end): modal appears when logo clicked on preview page

Modal appears when logo clicked on preview page

gargi-bansal committed:
feat(front end): fixed plus image styling

Fixed plus image styling

michellemtan reviewed (commented):
![image](https://user-images.githubusercontent.com/37464749/94862811-fc08b780-0495-11eb-9e3c-71a68c9ae93d.png)

Not sure if this should be part of this PR, but currently when the user hasn't selected any cards, the Preview page is blank

But when the user does the following:
1. Goes to /preview page by url and presses back button
2. Currently completed cards: 0/10, selects a card, currently completed cards updates to: 1/10
3. Next button doesn't work but back button acts as a next button?

gargi-bansal commented:
> ![image](https://user-images.githubusercontent.com/37464749/94862811-fc08b780-0495-11eb-9e3c-71a68c9ae93d.png)
> 
> Not sure if this should be part of this PR, but currently when the user hasn't selected any cards, the Preview page is blank
> 
> But when the user does the following:
> 
> 1. Goes to /preview page by url and presses back button
> 2. Currently completed cards: 0/10, selects a card, currently completed cards updates to: 1/10
> 3. Next button doesn't work but back button acts as a next button?

@michellemtan I can try make it automatically redirect to another page if no cards are visible (atm I'm only showing the cards that have been filled in). Tbh I can just show all of them if that's better and have them (greyed out/disabled) except I think the slider position always sets itself to 0 for those which is incorrect, hence why I filtered them out.

The second issue about going back is actually working properly. Going back on the preview page takes you back to the last card in DisplayCards. So you should only be able to go back from there. Imagine if you had filled out all the cards and navigated to preview and then want to go back again. I think it's best for the last card to be displayed.

gargi-bansal committed:
Merge branch 'master' into hrt-62-preview-ui-changes

mitchfaulcon commented:
@gargi-bansal If they directly navigate to the /Preview URL, I don't think the chart would be associated with any course so would it be best to redirect to the dashboard?

gargi-bansal commented:
@mitchfaulcon I sort of want to add this as a bug on the backlog, merge my branch in and wait for the UI/backend linking to be done first. That way I can see what we store in state and how it would be easiest to detect whether we already have a chart. Atm I don't know if we're using state or will be putting the chart ID in the URL.

gargi-bansal committed:
Merge branch 'master' into hrt-62-preview-ui-changes

mitchfaulcon commented:
> @mitchfaulcon I sort of want to add this as a bug on the backlog, merge my branch in and wait for the UI/backend linking to be done first. That way I can see what we store in state and how it would be easiest to detect whether we already have a chart. Atm I don't know if we're using state or will be putting the chart ID in the URL.

Sounds good to me

michellemtan commented:
> > ![image](https://user-images.githubusercontent.com/37464749/94862811-fc08b780-0495-11eb-9e3c-71a68c9ae93d.png)
> > Not sure if this should be part of this PR, but currently when the user hasn't selected any cards, the Preview page is blank
> > But when the user does the following:
> > 
> > 1. Goes to /preview page by url and presses back button
> > 2. Currently completed cards: 0/10, selects a card, currently completed cards updates to: 1/10
> > 3. Next button doesn't work but back button acts as a next button?
> 
> @michellemtan I can try make it automatically redirect to another page if no cards are visible (atm I'm only showing the cards that have been filled in). Tbh I can just show all of them if that's better and have them (greyed out/disabled) except I think the slider position always sets itself to 0 for those which is incorrect, hence why I filtered them out.
> 
> The second issue about going back is actually working properly. Going back on the preview page takes you back to the last card in DisplayCards. So you should only be able to go back from there. Imagine if you had filled out all the cards and navigated to preview and then want to go back again. I think it's best for the last card to be displayed.

Re the first issue: Either works! I think it's just better than having an empty page because I thought something was broken 👍 
Re the second issue: Ohhh okay that makes sense, I think once we get the proper dimensions and cards from Claire, it'll be more obvious to navigate backwards!

gargi-bansal commented:
> > > ![image](https://user-images.githubusercontent.com/37464749/94862811-fc08b780-0495-11eb-9e3c-71a68c9ae93d.png)
> > > Not sure if this should be part of this PR, but currently when the user hasn't selected any cards, the Preview page is blank
> > > But when the user does the following:
> > > 
> > > 1. Goes to /preview page by url and presses back button
> > > 2. Currently completed cards: 0/10, selects a card, currently completed cards updates to: 1/10
> > > 3. Next button doesn't work but back button acts as a next button?
> > 
> > 
> > @michellemtan I can try make it automatically redirect to another page if no cards are visible (atm I'm only showing the cards that have been filled in). Tbh I can just show all of them if that's better and have them (greyed out/disabled) except I think the slider position always sets itself to 0 for those which is incorrect, hence why I filtered them out.
> > The second issue about going back is actually working properly. Going back on the preview page takes you back to the last card in DisplayCards. So you should only be able to go back from there. Imagine if you had filled out all the cards and navigated to preview and then want to go back again. I think it's best for the last card to be displayed.
> 
> Re the first issue: Either works! I think it's just better than having an empty page because I thought something was broken 👍
> Re the second issue: Ohhh okay that makes sense, I think once we get the proper dimensions and cards from Claire, it'll be more obvious to navigate backwards!

@michellemtan As mentioned in the previous comment, I'd rather add this as a bug and wait till the backend is linked up before fixing this. Knowing what is stored in state/URL will make it easier to figure out how to solve the issue tidily.

michellemtan commented:
> > > > ![image](https://user-images.githubusercontent.com/37464749/94862811-fc08b780-0495-11eb-9e3c-71a68c9ae93d.png)
> > > > Not sure if this should be part of this PR, but currently when the user hasn't selected any cards, the Preview page is blank
> > > > But when the user does the following:
> > > > 
> > > > 1. Goes to /preview page by url and presses back button
> > > > 2. Currently completed cards: 0/10, selects a card, currently completed cards updates to: 1/10
> > > > 3. Next button doesn't work but back button acts as a next button?
> > > 
> > > 
> > > @michellemtan I can try make it automatically redirect to another page if no cards are visible (atm I'm only showing the cards that have been filled in). Tbh I can just show all of them if that's better and have them (greyed out/disabled) except I think the slider position always sets itself to 0 for those which is incorrect, hence why I filtered them out.
> > > The second issue about going back is actually working properly. Going back on the preview page takes you back to the last card in DisplayCards. So you should only be able to go back from there. Imagine if you had filled out all the cards and navigated to preview and then want to go back again. I think it's best for the last card to be displayed.
> > 
> > 
> > Re the first issue: Either works! I think it's just better than having an empty page because I thought something was broken 👍
> > Re the second issue: Ohhh okay that makes sense, I think once we get the proper dimensions and cards from Claire, it'll be more obvious to navigate backwards!
> 
> @michellemtan As mentioned in the previous comment, I'd rather add this as a bug and wait till the backend is linked up before fixing this. Knowing what is stored in state/URL will make it easier to figure out how to solve the issue tidily.

My bad, I didn't look at your reply to Mitchell's comment! Sounds good to me
